File: Remove Duplicates from Sorted List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        if (head == NULL) return NULL;
        
        ListNode *p, *q;
        p = head;
        while (p->next) {
            q = p->next;
            if (p->val == q->val) {
                p->next = q->next;
                delete q;
            } else {
                p = p->next;
            }
            
        }
        
        return head;
    }
};

File: Remove Duplicates from Sorted List.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode DeleteDuplicates(ListNode head) {
        if (head?.next == null) {
            return head;
        }
        
        var p = head;
        
        while (p != null) {
            var q = p.next;
            while (p.val == q?.val) {
                q = q.next;
            }
            if (p.next != q) {
                p.next = q;
            }
            p = q;
        }
        
        return head;
    }
}
